ibm/service/secretsmanager/resource_ibm_sm_configuration_private_certificate_root_CA.go

```diff
@@ -0,0 +1,724 @@
+				ForceNew:    true,
+				Description: "The number of bits to use to generate the private key.Allowable values for RSA keys are: `2048` and `4096`. Allowable values for EC keys are: `224`, `256`, `384`, and `521`. The default for RSA keys is `2048`. The default for EC keys is `256`.",
+			},
+			"max_path_length": &schema.Schema{
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "The maximum path length to encode in the generated certificate. `-1` means no limit.If the signing certificate has a maximum path length set, the path length is set to one less than that of the signing certificate. A limit of `0` means a literal path length of zero.",
+			},
+			"exclude_cn_from_sans": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Controls whether the common name is excluded from Subject Alternative Names (SANs).If the common name set to `true`, it is not included in DNS or Email SANs if they apply. This field can be useful if the common name is a human-readable identifier, instead of a hostname or an email address.",
+			},
+			"permitted_dns_domains": &schema.Schema{
+				Type:        schema.TypeList,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The allowed DNS domains or subdomains for the certificates that are to be signed and issued by this CA certificate.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"ou": &schema.Schema{
+				Type:        schema.TypeList,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The Organizational Unit (OU) values to define in the subject field of the resulting certificate.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"organization": &schema.Schema{
+				Type:        schema.TypeList,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The Organization (O) values to define in the subject field of the resulting certificate.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"country": &schema.Schema{
+				Type:        schema.TypeList,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The Country (C) values to define in the subject field of the resulting certificate.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"locality": &schema.Schema{
+				Type:        schema.TypeList,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The Locality (L) values to define in the subject field of the resulting certificate.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"province": &schema.Schema{
+				Type:        schema.TypeList,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The Province (ST) values to define in the subject field of the resulting certificate.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"street_address": &schema.Schema{
```

Comment:
In every resource we do have some arguments which does't support upate/patch operation should those kind of optional attribites marked as force_new or diff suppreess (suppress the change made by user)

Reply:
The decision was to use force_new so that the user is aware that this change cannot be performed without TF replacing the resource. We want the config to reflect the actual state of the resource, and that is achieved by the force_new marking. Diff suppress may cause misunderstandings about the true state of the resource